compiler/src/dmd/dtemplate.d

```diff
@@ -6261,27 +6261,40 @@ extern (C++) class TemplateInstance : ScopeDsymbol
             return false;
         }
 
+        // When no errors instantiating, this should only be called on
+        // the primary instantiation.
+        assert(this is inst || inst is null);
```

Comment:
Wow, so `inst` can be null?!

Reply:
Yes https://github.com/dlang/dmd/blob/314b20cd68c80f2a9fe4bf7ac318dd1b62618ccb/compiler/src/dmd/dtemplate.d#L892-L902

Reply:
OK, `inst` is `null` if it comes from a 'dummy' instance to evaluate a constraint.